q2_diversity/_beta/_visualizer.py

```diff
@@ -40,17 +40,25 @@ def bioenv(output_dir: str, distance_matrix: skbio.DistanceMatrix,
     metadata = metadata.filter_columns(column_type='numeric')
     non_numeric_cols = pre_filtered_cols - set(metadata.columns)
 
+    # Drop samples that have any missing values.
+    # TODO use Metadata API if this type of filtering is supported in the
+    # future.
+    # NOTE this whole cast to df to dropnans then cast back to metadata for
+    # other filtering then go back to df again to use is gross. It doesn't look
+    # like metadata currently supports this type of filtering (as per the
+    # above TODO), but this sure makes me wish it did.
+    df = metadata.to_dataframe()
+    df = df.dropna(axis='index', how='any')
+
+    metadata = qiime2.Metadata(df)
```

Comment:
@thermokarst I just wanted to clean up the comments here before merging.  Shall I go ahead and tidy up this PR myself, or wait until Anthony is back in action?

```suggestion
    # Drop samples that have any missing values.
    # TODO use Metadata API if more filtering is supported in the future.
    df = metadata.to_dataframe()
    df = df.dropna(axis='index', how='any')
    metadata = qiime2.Metadata(df)
```

Reply:
Thanks for checking! I checked in with @Oddant1 earlier this week - he said to go ahead and push up any edits necessary to close up his PRs. Merge away!